### ISSUE TITLE
Missing alt text for IPython.display.Image

### ISSUE THREAD

**Issue by psychemedia** [ipython/ipython#13577]
Image alt text when rendering `IPython.display.Image` is not displayed when `Image()` is created from `filename` or `data` elements. The [`test_image_alt_tag()`](https://github.com/ipython/ipython/blob/b7f20475905c16160c2e9e6ccbce817a0f1c3493/IPython/core/tests/test_display.py#L489) only tests `Image()` objects created from `url`).

The following (included in tests) passes:

```python
import nose.tools as nt

thisurl = "http://example.com/image.png"
img = Image(url=thisurl, alt="an image")
nt.assert_equal(u'<img src="%s" alt="an image"/>' % (thisurl), img._repr_html_())
```

The following two cases *do* render an image but fail to produce alt text (indeed, they don't seem to return `__repr_html__()`?

```python
from numpy import random
import matplotlib.pyplot as plt

x = random.randint(10, size=(10))
y = random.randint(10, size=(10))

plt.scatter(x, y)
img_fn = 'example.png'
plt.savefig(img_fn)

img2 = Image(filename=img_fn, alt="another image")
# _repr_html_() returns None
assert img2._repr_html_() == None

nt.assert_equal(u'<img src="%s" alt="another image"/>' % (img_fn), img2._repr_html_())

"""
AssertionError: '<img src="example.png" alt="another image"/>' != None
"""
```

and:

```python
with open(img_fn, "rb") as image:
    img_bytes = bytearray( image.read() )
    
    img3 = Image(filename=img_fn, alt="another image")
    repr_html = img3._repr_html_()
    
    # _repr_html_() returns None
    assert img3._repr_html_() == None
```

**Comment by psychemedia** [comment]
alt text is only set on `<img>` tag if we are not embedding image: https://github.com/ipython/ipython/commit/24aa3931604b1421becdf615d614d5e3ae2aeee6#diff-744c8a0c7792b132c2bbde8802d8c5234175720ddb921111c57b4aa2cac673bcL964

If we are embedding, a mime-type bundle is returned and no alt text ia set.

**Comment by MrMino** [comment]
Yes, without the `embed` argument data read from files is assumed to be rendered without usage of `_repr_html_` (_as far as I can tell_ - this code was created in 2011...).

> If we are embedding, a mime-type bundle is returned and no alt text is set.
Are you sure? Can you give your repro steps wrt this?

![image](https://user-images.githubusercontent.com/6691643/158718586-d1ee45c6-bd5a-4d6b-9a33-b1a22f3effec.png)

**Comment by psychemedia** [comment]
The issue is maybe in rendering elsewhere then? I spotted this in Jupyter notebook classic.

For example, alt text from `url=` works fine:

<img width="716" alt="image" src="https://user-images.githubusercontent.com/82988/158776375-25907067-cceb-4948-b9bb-1462c687aa1e.png">

But with the embedded image, no alt text is rendered:

<img width="663" alt="image" src="https://user-images.githubusercontent.com/82988/158776635-a0b1ed6d-c2e5-43d6-a5c5-d5db736666b3.png">

**Comment by psychemedia** [comment]
Some examples of alt text only rendering under limited circumstances:

```python
# Alt text and image renders
img = Image(url='http://example.com/example.png', alt='an image', embed=False)
assert img._repr_html_() == '<img src="http://example.com/example.png" alt="an image"/>'

# Alt text renders but image is lost
img = Image(filename='example.png', alt='an image', embed=False)
assert img._repr_html_() == '<img src="None" alt="an image"/>'

# Alt text doesn't render

img = Image(filename='example.png', alt='an image', embed=True)
assert img._repr_html_()==None

```

**Comment by camchenry** [comment]
I'd like to revive this thread, as I'm able to reproduce the issue when rendering Jupyter notebooks on GitHub (which uses `nbconvert` to do the rendering). These examples _do_ work for rendering `alt` text correctly in the web page:

```python
from IPython.display import Image, HTML

# Example 1: Using `Image` with `embed=False`
Image(filename='plot.png', alt='A scatter plot showing the relationship between temperature and ice cream sales', embed=False)

# Example 2: Using HTML output manually
html_img = '<img src="data_visualization.png" alt="Line graph displaying website traffic over a 12-month period, showing seasonal trends with peaks in November and December">'
HTML(html_img)
```

These examples do not work:

```python
# Example 3: Using `Image` by default
Image(filename='plot.png', alt='A scatter plot showing the relationship between temperature and ice cream sales')

# Example 4: Using `Image` with `embed=True`
Image(filename='plot.png', alt='A scatter plot showing the relationship between temperature and ice cream sales', embed=True)

# Example 5: Using `matplotlib`
import matplotlib.pyplot as plt
import numpy as np
fig, ax = plt.subplots()
x = np.linspace(0, 10, 100)
y = np.sin(x)
ax.plot(x, y)
ax.set_title('Sine Wave')
ax.set_xlabel('x')
ax.set_ylabel('sin(x)')
plt.show()
```

I believe the `matplotlib` issue around `_repr_html_` is captured here: https://github.com/matplotlib/matplotlib/issues/15971, but I think the `Image` examples are something relevant to this issue.
